Adding Cuda Variant to SW4Lite

rspavel opened the pull request:
Added cuda variant of sw4lite  as per guidance in SW4Lite README

rspavel committed:
Adding Cuda Variant to SW4Lite

Added cuda variant of sw4lite  as per guidance in README

adamjstewart commented on var/spack/repos/builtin/packages/sw4lite/package.py:
```diff
@@ -52,6 +52,14 @@ def build_targets(self):
+        if '+cuda' in self.spec:
+            targets.append('NVCC = {0}'.format(
+                self.spec['cuda'].prefix.bin.nvcc))
+            targets.append('HOSTCOMP = {0}'.format(self.compiler.cxx))
```
@ax3l @svenevs should this be `self.compiler.cxx` (the actual compiler) or `spack_cxx` (the compiler wrapper)?

svenevs replied:
Hmm. I'm not really sure, I guess it feels more spack-like to make both the wrappers? I don't know of any other examples that need to do this though, I think everything else just relies on things being (injected) in the PATH?

adamjstewart commented on var/spack/repos/builtin/packages/sw4lite/package.py:
```diff
@@ -52,6 +52,14 @@ def build_targets(self):
+            targets.append('HOSTCOMP = {0}'.format(self.compiler.cxx))
+            targets.append('MPIPATH= {0} '.format(self.spec['mpi'].prefix))
+            targets.append('gpuarch= {0}'.format(self.cuda_flags(cuda_arch)))
+            targets.append('MPIINC = {0}'.format(self.spec['mpi'].include))
```
Did you mean `prefix.include`? It might be best to use `headers.directories[0]` here.

rspavel replied:
Ooh, did not know that was a thing. That will make a lot of other things a lot easier. Fixed and thanks.

rspavel committed:
Updated SW4Lite+cuda to Current Header Conventions

Updated sw4lite+cuda to use current conventions for spackage include
dirs

rspavel committed:
Fixing FLake8 Issue with Sw4lite+cuda Fix

Fixed overly long line and further underlined sticky note reminding me
to run flake8 BEFORE pushing

rspavel committed:
Switching to Spack Compiler Wrapper

Switching to spack compiler wrapper for consistency
